src/NuGet.Clients/NuGet.CommandLine/Commands/SearchCommand.cs

```diff
@@ -0,0 +1,208 @@
+                    .Select(s => PackageSourceProviderExtensions.ResolveSource(configurationSources, s))
+                    .ToList();
+            }
+            else
+            {
+                packageSources = configurationSources;
+            }
+            return packageSources;
+        }
+
+        public override async Task ExecuteCommandAsync()
+        {
+            ILogger logger = Console;
+            CancellationToken cancellationToken = CancellationToken.None;
+
+            SearchFilter searchFilter = new SearchFilter(includePrerelease: PreRelease);
+
+            IList<PackageSource> listEndpoints = GetEndpointsAsync();
+
+            foreach (PackageSource source in listEndpoints)
+            {
+                string target = source.Source;
+                string name = source.Name;
+
+                SourceRepository repository = Repository.Factory.GetCoreV3(target);
+                PackageSearchResource resource = await repository.GetResourceAsync<PackageSearchResource>();
+
+                if (resource is null)
+                {
+                    continue;
+                }
+
+                IEnumerable<IPackageSearchMetadata> results = await resource.SearchAsync(
+                    string.Join("+", Arguments).Trim().Replace(' ', '+'), // The arguments (query strings) are joined with '+' so that the queries are passed to the
+                                                                          // search query url with the correct format (eg. "logging extensions" --> "logging+extensions")
+                    searchFilter,
+                    skip: 0,
+                    take: Take,
+                    logger,
+                    cancellationToken);
+
+                Console.WriteLine(new string('=', 20));
+                Console.WriteLine($"Source: {source.Name}");
+
+                if (!results.Any())
+                {
+                    Console.WriteLine(new string('-', 20));
+                    Console.WriteLine("No results found.");
+                    Console.WriteLine(new string('-', 20) + "\n");
+                }
+                else
+                {
+                    if (Verbosity == Verbosity.Quiet)
+                    {
+                        System.Console.WriteLine($"Source: {source.Name}");
+                        System.Console.WriteLine(new string('-', 20));
+                    }
+                    PrintResults(results);
+                }
+            }
+        }
+
+        private void PrintResults(IEnumerable<IPackageSearchMetadata> results)
+        {
+            foreach (IPackageSearchMetadata result in results)
+            {
+                Console.WriteLine(new string('-', 20));
+
+                CultureInfo culture = CultureInfo.CurrentCulture;
+
+                string printBasicInfo = $"> {result.Identity.Id} | {result.Identity.Version.ToNormalizedString()}";
+
+                string downloads, printDownloads;
```

Comment:
I'd suggest refactoring these to instead use `StringBuilder`, then change your concatenation below to use the `sb.Append(...)`.